Detectors/TRD/calibration/src/PadCalibCCDBBuilder.cxx

```diff
@@ -0,0 +1,551 @@
+        coordinates.push_back(irow);
+        emptyBins.push_back(coordinates);
+      }
+    } // loop over columns
+  }   // loop over rows
+  return emptyBins;
+}
+
+std::vector<std::vector<int>> PadCalibCCDBBuilder::findInhomogeneities(TH2F* hDet, float allowedDifference)
+{ // finds bins that have 1+ neighbour with significantly different content
+  // gives the coordonates of the pair
+  std::vector<std::vector<int>> suspiciousBinPairs;
+
+  // check for edges - this operation goes before filling
+  // edges along X:
+  int xFirst = hDet->FindFirstBinAbove();
+  int xLast = hDet->FindLastBinAbove(); // stop the process at penultimate vs ultimate row/column
+  // // edges along Y:
+  int yFirst = hDet->FindFirstBinAbove(0, 2);
+  int yLast = hDet->FindLastBinAbove(0, 2); // stop the process at penultimate vs ultimate row/column
+
+  // int nbins_x = hDet->GetNbinsX();
+  // int nbins_y = hDet->GetNbinsY();
+  for (int irow = 0; irow < hDet->GetNbinsY(); irow++) {
+
+    int thisRow = hDet->GetXaxis()->FindBin(irow);
+    // if( thisRow < yFirst || thisRow > yLast ) continue;
+
+    for (int icolumn = 0; icolumn < hDet->GetNbinsX(); icolumn++) {
+
+      int thisColumn = hDet->GetXaxis()->FindBin(icolumn);
+      // if( thisColumn < xFirst || thisColumn > xLast ) continue;
+
+      std::vector<int> pair = compareGain(hDet, thisColumn, thisRow, 1, 0, allowedDifference);
+      if (!(std::any_of(pair.begin(), pair.end(), [](int i) { return i == -1; }))) {
+        suspiciousBinPairs.push_back(pair);
+      }
+      pair.clear();
+
+      pair = compareGain(hDet, thisColumn, thisRow, 0, 1, allowedDifference);
+      if (!(std::any_of(pair.begin(), pair.end(), [](int i) { return i == -1; }))) {
+        suspiciousBinPairs.push_back(pair);
+      }
+
+    } // loop over columns
+
+  } // loop over rows
+
+  return suspiciousBinPairs;
+}
+
+float PadCalibCCDBBuilder::getAverageFromNeighbors(TH2F* hDet, int column, int row, int nbuffer)
+{ // takes bin (column,row) from hDet map
+  // and averages the gain in its (up to) 8 neighbours
+  float average = 0.;
+  std::vector<float> gainNeighbor;
+  int offset = nbuffer / 2;
+
+  for (int irow = 0; irow < nbuffer; irow++) {
+
+    int rowNeighbor = (row - offset) + irow;
+    if (rowNeighbor < 0 || rowNeighbor >= hDet->GetNbinsY())
+      continue; // avoids under and overflow
+    for (int icol = 0; icol < nbuffer; icol++) {
+      if (icol == 1 && irow == 1)
+        continue; // exclude self
+      int colNeighbor = (column - offset) + icol;
+      if (colNeighbor < 0 || colNeighbor >= hDet->GetNbinsX())
+        continue; // avoids under and overflow
+      float tempGain = TMath::Abs(hDet->GetBinContent(hDet->GetXaxis()->FindBin(colNeighbor), hDet->GetYaxis()->FindBin(rowNeighbor)));
+      if (tempGain <= 0)
+        continue; // exclude empty/negative bins
+      gainNeighbor.push_back(tempGain);
+    }
+  }
+  int numberOfNeighbors = (int)gainNeighbor.size();
```

Comment:
It looks like you have a few type casts from sizes to signed integers. Try to avoid them by using size_t (or auto) as the type.